File: testprojects/csharp/benchmark/infrastructure/socialmedia/locationbased/latitude/CheckInDAO.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace CSharpBenchmark.infrastructure.socialmedia.locationbased.latitude
{

    //Functional requirement 3.2
    //Test case 121: Class domain.locationbased.latitude.CheckIn is not allowed to use class infrastructure.socialmedia.locationbased.foursquare.CheckInDAO 
    //Result: TRUE
    public class CheckInDAO
    {
        public static String currentLocation = "Nijenoord 1, Utrecht";
    }
}

File: testresources/java/benchmark_2012/infrastructure/socialmedia/gowalla/CheckInDAO.java
package infrastructure.socialmedia.gowalla;

import domain.gowalla.CheckIn;
//Functional requirement 3.2.1
//Test case 161: The classes in package infrastructure.socialmedia.gowalla are not allowed to use modules in a higher layer
//Result: FALSE
public class CheckInDAO{
	//FR5.5
	private CheckIn checkin;

	public CheckInDAO(){
		//FR5.1
		System.out.println(checkin.getType());
	}
}